src/pipelines/floodscan_pipeline.py

```diff
@@ -0,0 +1,370 @@
+import os
+import re
+import shutil
+from datetime import datetime, timedelta
+from fileinput import filename
+from zipfile import ZipFile
+
+import numpy as np
+import pandas as pd
+import requests
+import xarray as xr
+from netCDF4.utils import ncinfo
+
+from ..utils import raster_utils
+from ..utils.azure_utils import download_from_azure, blob_client
+from ..utils.raster_utils import invert_lat_lon, create_date_range
+from .pipeline import Pipeline
+
+
+class FloodScanPipeline(Pipeline):
+    def __init__(self, **kwargs):
+        raw_path = kwargs["raw_path"]
+        processed_path = kwargs["processed_path"]
+        kwargs["metadata"]["version"] = kwargs["version"]
+
+        super().__init__(
+            container_name=kwargs["container_name"],
+            raw_path=raw_path,
+            processed_path=processed_path,
+            log_level=kwargs["log_level"],
+            mode=kwargs["mode"],
+            metadata=kwargs["metadata"],
+            use_cache=kwargs["use_cache"],
+        )
+
+        self.start_date = datetime.strptime(kwargs["start_date"], "%Y%m%d")
+        self.end_date = datetime.strptime(kwargs["end_date"], "%Y%m%d")
+        self.is_update = kwargs["is_update"]
+        self.is_full_historical_run = kwargs["is_full_historical_run"]
+        self.version = kwargs["version"]
+        self.sfed_historical = kwargs["sfed_historical"]
+        self.mfed_historical = kwargs["mfed_historical"]
+        self.sfed_base_url = os.getenv("FLOODSCAN_SFED_URL")
+        self.mfed_base_url = os.getenv("FLOODSCAN_MFED_URL")
+
+
+    def _generate_raw_filename(self, date, type):
+        return f"aer_floodscan_{type}_area_flooded_fraction_africa_90days_{date.strftime('%Y%m%d')}.zip"
+
+    def _generate_unzipped_filename(self, date, type):
+        return f"aer_{type}_area_300s_{date.strftime('%Y%m%d')}_v0{self.version}r01.tif"
+
+    def _generate_processed_filename(self, date):
+        return f"aer_area_300s_{date.strftime('%Y%m%d')}_v0{self.version}r01.tif"
```

Comment:
Could we store the date as `'%Y-%m-%d'` for consistency with the other processed files?